lib/mqtt/mqtt_client.dart

```diff
@@ -0,0 +1,268 @@
+import "dart:convert";
+import "dart:typed_data";
+import "package:mqtt_client/mqtt_client.dart";
+import "package:mqtt_client/mqtt_server_client.dart";
+import "package:noheva_visitor_ui/mqtt/model/status_message.dart";
+import "package:simple_logger/simple_logger.dart";
+import "package:typed_data/typed_buffers.dart";
+import "../main.dart";
+import "../utils/device_info.dart";
+
+/// MQTT Client
+class MqttClient {
+  String? _deviceId;
+  MqttServerClient? _client;
+
+  Future<String> get _statusTopic async {
+    var deviceId = _deviceId;
+
+    if (deviceId == null) {
+      SimpleLogger().warning("Device ID not found, cannot get status topic.");
+
+      return "";
+    }
+
+    return "noheva/$environment/$deviceId/status";
+  }
+
+  bool get isConnected =>
+      _getClientConnectionStatus() == MqttConnectionState.connected.name;
+
+  Map<String, Function(String)> listeners = {};
+
+  /// Connects MQTT server using [deviceId] as client id if not already connected.
+  Future<void> connect(String deviceId) async {
+    _deviceId = deviceId;
+    var client = _initializeClient(deviceId);
+
+    var mqttUsername = configuration.getMqttUsername();
+    var mqttPassword = configuration.getMqttPassword();
+    if (client.connectionStatus!.state == MqttConnectionState.connected) {
+      SimpleLogger().info("MQTT Client already connected");
+
+      return;
+    }
+
+    if (await DeviceInfo.getSerialNumber() == null) {
+      SimpleLogger().warning("Device ID not found, cannot connect to MQTT.");
+
+      return;
+    }
```

Comment:
Return value of `DeviceInfo.getSerialNumber()` is already passed to this `connect()` method as a parameter and this same check is already done in `main.dart`

Reply:
removed